docs/bnb-greenfield/for-developers/cross-chain-integration/demo-contract-as-bucket-owner.md

```diff
@@ -0,0 +1,127 @@
+---
+title: Contract as Bucket Owner - BNB Greenfield Demo
+---
+
+#  Greenfield Demo: Contract as Bucket Owner
+BNB Greenfield provides the ability for a contract to manage buckets through greenfield cross-chain transactions, 
+and we'll use a demo to briefly demonstrate the process.
+
+## Source Code
+- Greenfield Demo App Contract: [GreenfieldDemo](https://github.com/bnb-chain/greenfield-contracts/blob/develop/contracts/example/GreenfieldDemo.sol)
+- Script to Interact with Demo: [Demo Script](https://github.com/bnb-chain/greenfield-contracts/blob/develop/scripts/11-demo-contract-approve-eoa-upload.ts) 
+
+## Greenfield Demo Contract
+The Demo includes the following parts:
+### greenfield contracts constant:
+```solidity
+    // BSC testnet
+    address public constant TOKEN_HUB = 0xED8e5C546F84442219A5a987EE1D820698528E04;
+    address public constant CROSS_CHAIN = 0xa5B2c9194131A4E0BFaCbF9E5D6722c873159cb7;
+    address public constant BUCKET_HUB = 0x5BB17A87D03620b313C39C24029C94cB5714814A;
+    address public constant PERMISSION_HUB = 0x25E1eeDb5CaBf288210B132321FBB2d90b4174ad;
+    address public constant SP_ADDRESS_TESTNET = 0x5FFf5A6c94b182fB965B40C7B9F30199b969eD2f;
+    address public constant GREENFIELD_EXECUTOR = 0x3E3180883308e8B4946C9a485F8d91F8b15dC48e;
+```
+The version of greenfield contracts deployed on BSC testnet is configured here.
+
+### create bucket and set bucket flow rate limit
+```solidity
+    function createBucket(string memory bucketName, uint256 transferOutAmount, bytes memory _executorData) external payable;
+```
+Provides the bucket name, initial BNB amount transferred to the demo contract on Greenfield and the executor data 
+to set bucket flow rate limit.
+
+### create policy to allow eoa account to upload files to the bucket
+```solidity
+    function createPolicy(bytes memory createPolicyData) external payable;
+```
```

Comment:
Construct the `createPolicyData` parameters to grant permission to the dedicated principle for the created bucket owned by the smart contract.

Reply:
fixed